sdk/synapse/azure-synapse-accesscontrol/azure/synapse/accesscontrol/_access_control_client.py

```diff
@@ -44,37 +47,63 @@ def __init__(
         self.role_assignments = RoleAssignmentsOperations(
-            self._client, self._config, self._serialize, self._deserialize)
+            self._client, self._config, self._serialize, self._deserialize
+        )
         self.role_definitions = RoleDefinitionsOperations(
-            self._client, self._config, self._serialize, self._deserialize)
+            self._client, self._config, self._serialize, self._deserialize
+        )
 
-    def _send_request(self, http_request, **kwargs):
+    def send_request(self, request, **kwargs):
         # type: (HttpRequest, Any) -> HttpResponse
         """Runs the network request through the client's chained policies.
 
-        :param http_request: The network request you want to make. Required.
-        :type http_request: ~azure.core.pipeline.transport.HttpRequest
-        :keyword bool stream: Whether the response payload will be streamed. Defaults to True.
+        We have helper methods to create requests specific to this service in `azure.synapse.accesscontrol.rest`.
+        Use these helper methods to create the request you pass to this method. See our example below:
+
+        >>> from azure.synapse.accesscontrol.rest import build_check_principal_access_request
+        >>> request = build_check_principal_access_request(json, content)
+        <HttpRequest [POST], url: '/checkAccessSynapseRbac'>
+        >>> response = client.send_request(request)
+        <HttpResponse: 200 OK>
+
+        For more information on this code flow, see https://aka.ms/azsdk/python/protocol/quickstart
+
+        For advanced cases, you can also create your own :class:`~azure.synapse.accesscontrol.core.rest.HttpRequest`
+        and pass it in.
+
+        :param request: The network request you want to make. Required.
+        :type request: ~azure.synapse.accesscontrol.core.rest.HttpRequest
+        :keyword bool stream: Whether the response payload will be streamed. Defaults to False.
         :return: The response of your network call. Does not do error handling on your response.
-        :rtype: ~azure.core.pipeline.transport.HttpResponse
+        :rtype: ~azure.synapse.accesscontrol.core.rest.HttpResponse
         """
+        request_copy = deepcopy(request)
         path_format_arguments = {
-            'endpoint': self._serialize.url("self._config.endpoint", self._config.endpoint, 'str', skip_quote=True),
+            "endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, "str", skip_quote=True),
         }
-        http_request.url = self._client.format_url(http_request.url, **path_format_arguments)
-        stream = kwargs.pop("stream", True)
-        pipeline_response = self._client._pipeline.run(http_request, stream=stream, **kwargs)
-        return pipeline_response.http_response
+        request_copy.url = self._client.format_url(request_copy.url, **path_format_arguments)
+        if kwargs.pop("stream", False):
+            return _StreamContextManager(
```

Comment:
Why we return the private type?

Reply:
This is kind of an old implementation of `azure.core.rest`, going to update this once we have solidified `azure.core.rest` more, so will eventually not be a private type.